src/Modal/index.ts

```diff
@@ -0,0 +1,123 @@
+import {Component} from 'react';
+import {render} from 'react-universal-interface';
+import FocusLock from 'react-focus-lock';
+import {h, isClient, on, off, noop} from '../util';
+import {Overlay, IOverlayProps} from '../Overlay';
+
+let id = 0;
+
+const ESC = 27;
+
+export interface IModalProps extends IOverlayProps {
+  onEsc?: (event: KeyboardEvent) => void;
+}
+
+export interface IModalState {
+  idTitle: string;
+  idDescription: string;
+}
+
+export class Modal extends Component<IModalProps, IModalState> {
+  id: number;
+  el: HTMLElement = null;
+  activeEl: Element; // Previous active element;
+
+  constructor (props, context) {
+    super(props, context);
+
+    this.id = id++;
+
+    this.state = {
+      idTitle: 'dialog-title-' + this.id,
+      idDescription: 'dialog-descr-' + this.id
+    };
+
+    this.activeEl = isClient ? document.activeElement : null;
+  }
+
+  componentDidMount () {
+    on(document, 'keydown', this.onKey);
+  }
+
+  componentWillUnmount () {
+    off(document, 'keydown', this.onKey);
+
+    const siblings = Array.from(document.body.children);
+
+    for (let i = 0; i < siblings.length; i++) {
+      const sibling = siblings[i] as HTMLElement;
+
+      if (sibling === this.el) {
+        continue;
+      }
+
+      if ((sibling as any).__modal_lock !== this) {
+        continue;
+      }
+
+      delete (sibling as any).__modal_lock;
+      (sibling as any).inert = false;
+      sibling.style.removeProperty('pointer-events');
+      sibling.style.removeProperty('user-select');
+      sibling.removeAttribute('aria-hidden');
+    }
+
+    // Focus previously active element.
+    if (this.activeEl && (this.activeEl as any).focus) {
+      (this.activeEl as any).focus();
+    }
+  }
+
+  onElement = (el) => {
+    this.el = el;
+
+    el.setAttribute('role', 'dialog');
+    el.classList.add('dialog');
+
+    el.setAttribute('aria-labelledby', 'dialog-title-' + this.id);
+    el.setAttribute('aria-describedby', 'dialog-descr-' + this.id);
+
+    const siblings = Array.from(document.body.children);
+
+    for (let i = 0; i < siblings.length; i++) {
+      const sibling = siblings[i] as HTMLElement;
+
+      if (sibling === el) {
+        continue;
+      }
+
+      if ((sibling as any).__modal_lock) {
+        continue;
+      }
+
+      (sibling as any).__modal_lock = this;
+      (sibling as any).inert = true;
+      sibling.style.setProperty('pointer-events', 'none');
```

Comment:
What if these elements already had a value for `pointer-events`, `user-select` or `aria-hidden`? It seems like that original value will be lost by opening and closing the the `Modal`.

Reply:
Yes, but the should not have any of those set.

Reply:
Please explain why not.

Reply:
A typical setup will be you will have two siblings at root level, one for your "app" and another one will be a modal:

```html
<body>
  <div id="app" />
  <div class="modal" />
</body>
```

You will render your React app into the first `<div>`

```html
render(<App />, document.getElementById('app'));
```

Modal will go into the second `<div>`.

Both `<div>`s will not have any attributes set.

If you are not using to React to build your app, you should not use this `<Modal>` component.

Reply:
Meanwhile, in the real world:

![image](https://user-images.githubusercontent.com/1030555/36434431-e94a88c4-1656-11e8-862d-4ed8709358b8.png)

Note: `aria-hidden="true"` on at least one of these.

Reply:
Maybe there should be some checking.